### PR TITLE
erlang 25.1

### PR DESCRIPTION
**Pull request by fishtreesugar**
Created with `brew bump-formula-pr`.

`resource` blocks may require updates.

### CODE REVIEW

**Comment by Porkepix** [comment]
Resource must be updated, and this can be migrated to `openssl@3`.

**Comment by carlocab** [comment]
> this can be migrated to `openssl@3`.

No, there are dependency conflicts.

**Comment by Porkepix** [comment]
> > this can be migrated to `openssl@3`.
> 
> No, there are dependency conflicts.

Yeah I see that, I migrated erlang 24 earlier, but this one have dependency conflicts. I guess those two formulas have to be migrated in the same time.

**Comment by fishtreesugar** [comment]
tsung's `openssl@1.1` dependence is introdueced by `erlang`, why it could conflicted?
```
$brew deps --tree tsung
tsung
├── erlang
│   ├── openssl@1.1
│   │   └── ca-certificates
│   └── wxwidgets
│       ├── jpeg-turbo
│       ├── libpng
│       └── libtiff
│           └── jpeg-turbo
└── gnuplot
    ├── gd
    │   ├── fontconfig
    │   │   └── freetype
    │   │       └── libpng
    │   ├── freetype
    │   │   └── libpng
    │   ├── jpeg-turbo
    │   ├── libavif
    │   │   ├── aom
    │   │   │   ├── jpeg-xl
    │   │   │   │   ├── brotli
    │   │   │   │   ├── giflib
    │   │   │   │   ├── imath
    │   │   │   │   ├── jpeg-turbo
    │   │   │   │   ├── libpng
    │   │   │   │   ├── openexr
    │   │   │   │   │   └── imath
    │   │   │   │   └── webp
    │   │   │   │       ├── giflib
    │   │   │   │       ├── jpeg-turbo
    │   │   │   │       ├── libpng
    │   │   │   │       └── libtiff
    │   │   │   │           └── jpeg-turbo
    │   │   │   └── libvmaf
    │   │   ├── jpeg-turbo
    │   │   └── libpng
    │   ├── libpng
    │   ├── libtiff
    │   │   └── jpeg-turbo
    │   └── webp
    │       ├── giflib
    │       ├── jpeg-turbo
    │       ├── libpng
    │       └── libtiff
    │           └── jpeg-turbo
    ├── libcerf
    ├── lua
    ├── pango
    │   ├── cairo
    │   │   ├── fontconfig
    │   │   │   └── freetype
    │   │   │       └── libpng
    │   │   ├── freetype
    │   │   │   └── libpng
    │   │   ├── glib
    │   │   │   ├── gettext
    │   │   │   └── pcre2
    │   │   ├── libpng
    │   │   ├── libx11
    │   │   │   ├── libxcb
    │   │   │   │   ├── libpthread-stubs
    │   │   │   │   │   └── pkg-config
    │   │   │   │   ├── libxau
    │   │   │   │   │   └── xorgproto
    │   │   │   │   └── libxdmcp
    │   │   │   │       └── xorgproto
    │   │   │   └── xorgproto
    │   │   ├── libxcb
    │   │   │   ├── libpthread-stubs
    │   │   │   │   └── pkg-config
    │   │   │   ├── libxau
    │   │   │   │   └── xorgproto
    │   │   │   └── libxdmcp
    │   │   │       └── xorgproto
    │   │   ├── libxext
    │   │   │   ├── libx11
    │   │   │   │   ├── libxcb
    │   │   │   │   │   ├── libpthread-stubs
    │   │   │   │   │   │   └── pkg-config
    │   │   │   │   │   ├── libxau
    │   │   │   │   │   │   └── xorgproto
    │   │   │   │   │   └── libxdmcp
    │   │   │   │   │       └── xorgproto
    │   │   │   │   └── xorgproto
    │   │   │   └── xorgproto
    │   │   ├── libxrender
    │   │   │   ├── libx11
    │   │   │   │   ├── libxcb
    │   │   │   │   │   ├── libpthread-stubs
    │   │   │   │   │   │   └── pkg-config
    │   │   │   │   │   ├── libxau
    │   │   │   │   │   │   └── xorgproto
    │   │   │   │   │   └── libxdmcp
    │   │   │   │   │       └── xorgproto
    │   │   │   │   └── xorgproto
    │   │   │   └── xorgproto
    │   │   ├── lzo
    │   │   └── pixman
    │   ├── fontconfig
    │   │   └── freetype
    │   │       └── libpng
    │   ├── freetype
    │   │   └── libpng
    │   ├── fribidi
    │   ├── glib
    │   │   ├── gettext
    │   │   └── pcre2
    │   └── harfbuzz
    │       ├── cairo
    │       │   ├── fontconfig
    │       │   │   └── freetype
    │       │   │       └── libpng
    │       │   ├── freetype
    │       │   │   └── libpng
    │       │   ├── glib
    │       │   │   ├── gettext
    │       │   │   └── pcre2
    │       │   ├── libpng
    │       │   ├── libx11
    │       │   │   ├── libxcb
    │       │   │   │   ├── libpthread-stubs
    │       │   │   │   │   └── pkg-config
    │       │   │   │   ├── libxau
    │       │   │   │   │   └── xorgproto
    │       │   │   │   └── libxdmcp
    │       │   │   │       └── xorgproto
    │       │   │   └── xorgproto
    │       │   ├── libxcb
    │       │   │   ├── libpthread-stubs
    │       │   │   │   └── pkg-config
    │       │   │   ├── libxau
    │       │   │   │   └── xorgproto
    │       │   │   └── libxdmcp
    │       │   │       └── xorgproto
    │       │   ├── libxext
    │       │   │   ├── libx11
    │       │   │   │   ├── libxcb
    │       │   │   │   │   ├── libpthread-stubs
    │       │   │   │   │   │   └── pkg-config
    │       │   │   │   │   ├── libxau
    │       │   │   │   │   │   └── xorgproto
    │       │   │   │   │   └── libxdmcp
    │       │   │   │   │       └── xorgproto
    │       │   │   │   └── xorgproto
    │       │   │   └── xorgproto
    │       │   ├── libxrender
    │       │   │   ├── libx11
    │       │   │   │   ├── libxcb
    │       │   │   │   │   ├── libpthread-stubs
    │       │   │   │   │   │   └── pkg-config
    │       │   │   │   │   ├── libxau
    │       │   │   │   │   │   └── xorgproto
    │       │   │   │   │   └── libxdmcp
    │       │   │   │   │       └── xorgproto
    │       │   │   │   └── xorgproto
    │       │   │   └── xorgproto
    │       │   ├── lzo
    │       │   └── pixman
    │       ├── freetype
    │       │   └── libpng
    │       ├── glib
    │       │   ├── gettext
    │       │   └── pcre2
    │       ├── graphite2
    │       └── icu4c
    ├── qt@5
    │   ├── freetype
    │   │   └── libpng
    │   ├── glib
    │   │   ├── gettext
    │   │   └── pcre2
    │   ├── jpeg-turbo
    │   ├── libpng
    │   ├── pcre2
    │   └── webp
    │       ├── giflib
    │       ├── jpeg-turbo
    │       ├── libpng
    │       └── libtiff
    │           └── jpeg-turbo
    └── readline
```

File: Formula/ejabberd.rb [hunk 1]
```diff
@@ -20,7 +21,7 @@ class Ejabberd < Formula
   depends_on "erlang"
   depends_on "gd"
   depends_on "libyaml"
-  depends_on "openssl@1.1"
+  depends_on "openssl@3"
```
Comment (cho-m) [inline Formula/ejabberd.rb:24]: For most formulae, we probably want to keep `openssl@1.1` until other changes are in place.

See:
- https://github.com/Homebrew/homebrew-core/issues/104784#issuecomment-1174378680
- https://github.com/Homebrew/homebrew-core/pull/85775#issuecomment-934666042

Comment (fishtreesugar) [reply Formula/ejabberd.rb]: just keep using `openssl@1.1`

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**